tests/ftests/071-sudo-set_default_systemd_cgroup.py

```diff
@@ -42,131 +36,144 @@ def prereqs(config):
     return result, cause
 
 
-def create_cgrp(config, CGNAME, controller=CONTROLLER, ignore_systemd=False):
-    global CGRPS_LIST, MODE
-
-    result = consts.TEST_PASSED
-    cause = None
-
-    cgrp = Cgroup(CGNAME, Version.CGROUP_V1)
-    if controller is not None:
-        cgrp.add_controller(controller)
-    cgrp.create()
-
-    if not CgroupCli.exists(config, controller, CGNAME, ignore_systemd=ignore_systemd):
-        result = consts.TEST_FAILED
-        if MODE == Mode.CGROUP_MODE_UNIFIED:
-            cause = 'Failed to create {}'.format(os.path.join('/sys/fs/cgroup', CGNAME))
-        else:
-            cause = (
-                        'Failed to create {}'
-                        ''.format(os.path.join('/sys/fs/cgroup', (controller or ''), CGNAME))
-                    )
-        return result, cause
-
-    CGRPS_LIST.append(cgrp)
-
-    return result, cause
-
-
 def setup(config):
-    global MODE
+    pass
+
 
+def test(config):
     result = consts.TEST_PASSED
     cause = None
 
-    # probe the current cgroup set up mode
-    MODE = Cgroup.cgroup_mode()
-
-    if not os.path.isdir(SYSTEMD_DEFAULT_CGROUP_DIR):
-        os.mkdir(SYSTEMD_DEFAULT_CGROUP_DIR)
+    #
+    # Test 1 - Ensure __set_default_systemd_cgroup() throws an exception if
+    #          libcgroup doesn't set a default (slice/scope) cgroup path
+    #
+    try:
+        Cgroup.__set_default_systemd_cgroup()
+    except RuntimeError as re:
+        if 'Failed to set' not in str(re):
+            result = consts.TEST_FAILED
+            cause = 'Expected \'Failed to set\' to be in the exception, ' \
+                    'received {}'.format(str(re))
+    else:
+        result = consts.TEST_FAILED
+        cause = '__set_default_systemd_cgroup() erroneously passed'
+
+    #
+    # Test 2 - write_default_systemd_scope() should succeed if the slice/scope
+    #          are invalid and we're not setting it as the default.  If we
+    #          create a cgroup at this point, it should be created at the root
+    #          cgroup level, and the default slice/scope should have no bearing
+    #
+    Cgroup.write_default_systemd_scope(SLICE, SCOPE, False)
+
+    pid = config.process.create_process(config)
+    cg = Cgroup(OTHER_CGNAME, Version.CGROUP_V2)
```

Comment:
Should the check for the `cgroup setup mode()` for `cgroup v2` be added in the `prereqs`, as the test case explicitly uses cgroup v2

Reply:
I don't think we need to limit this test to cgroup v2 only.  When a Cgroup class is instantiated via `Cgroup('TomsCgroup', Version.CGROUP_V2)`, the user is telling libcgroup that they will be speaking v2 format.  The library can/will convert the cgroup to v1 or hybrid if the user requests as such.

And my latest test run shows that this successfully runs on v1, hybrid, and v2:
https://github.com/drakenclimber/libcgroup/actions/runs/5271801648/jobs/9533192309

Reply:
True, sorry for the noise.